modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosKeywordReader.scala

```diff
@@ -0,0 +1,211 @@
+
+package seqexec.server.gmos
+
+import cats.Applicative
+import cats.effect.LiftIO
+import cats.effect.Sync
+import cats.implicits._
+import seqexec.server.ConfigUtilOps._
+import seqexec.server.keywords._
+import seqexec.server.gmos.GmosEpics.RoiStatus
+import edu.gemini.spModel.config2.Config
+import edu.gemini.spModel.data.YesNoType
+import edu.gemini.spModel.gemini.gmos.InstGmosCommon.IS_MOS_PREIMAGING_PROP
+import edu.gemini.spModel.seqcomp.SeqConfigNames.INSTRUMENT_KEY
+
+final case class RoiValues(xStart: Int, xSize: Int, yStart: Int, ySize: Int)
+
+final case class GmosObsKeywordsReader[F[_]: Sync](config: Config) {
+  private implicit val BooleanDefaultValue = DefaultHeaderValue.FalseDefaultValue
+
+  def preimage: F[Boolean] =
+    Sync[F].delay(
+        config
+          .extractAs[YesNoType](INSTRUMENT_KEY / IS_MOS_PREIMAGING_PROP)
+          .getOrElse(YesNoType.NO))
+      .map(_.toBoolean)
+      .safeValOrDefault
+}
+
+trait GmosKeywordReader[F[_]] {
+  def ccName:                    F[String]
+  // TODO Add NOD*
+  /*def nodMode:                 F[String]
+  def nodPix:                    F[Int]
+  def nodCount:                  F[Int]
+  def nodAxOff:                  F[Int]
+  def nodAyOff:                  F[Int]
+  def nodBxOff:                  F[Int]
+  def nodByOff:                  F[Int]*/
+  def maskId:                    F[Int]
+  def maskName:                  F[String]
+  def maskType:                  F[Int]
+  def maskLoc:                   F[Int]
+  def filter1:                   F[String]
+  def filter2:                   F[String]
+  def filter1Id:                 F[Int]
+  def filter2Id:                 F[Int]
+  def grating:                   F[String]
+  def gratingId:                 F[Int]
+  def gratingWavelength:         F[Double]
+  def gratingAdjustedWavelength: F[Double]
+  def gratingOrder:              F[Int]
+  def gratingTilt:               F[Double]
+  def gratingStep:               F[Double]
+  def dtaX:                      F[Double]
+  def dtaY:                      F[Double]
+  def dtaZ:                      F[Double]
+  def dtaZst:                    F[Double]
+  def dtaZen:                    F[Double]
+  def dtaZme:                    F[Double]
+  def stageMode:                 F[String]
+  def adcMode:                   F[String]
+  def dcName:                    F[String]
+  def detectorType:              F[String]
+  def detectorId:                F[String]
+  def exposureTime:              F[Double]
+  def adcUsed:                   F[Int]
+  def adcPrismEntSt:             F[Double]
+  def adcPrismEntEnd:            F[Double]
+  def adcPrismEntMe:             F[Double]
+  def adcPrismExtSt:             F[Double]
+  def adcPrismExtEnd:            F[Double]
+  def adcPrismExtMe:             F[Double]
+  def adcWavelength1:            F[Double]
+  def adcWavelength2:            F[Double]
+  def detNRoi:                   F[Int]
+  def roiValues:                 F[List[(Int, RoiValues)]]
```

Comment:
Instead of a `Map` of actions we'll get a fully resolved action with a map of values. I switched to a `List` of tuples to ensure we preserve the order